middleware.py

```
@@ -0,0 +1,111 @@
+
+# ── CSRF (double-submit cookie, zero JS) ─────────────────────────
+
+CSRF_COOKIE = "_csrf"
+CSRF_FIELD = "csrf_token"
+SAFE_METHODS = {"GET", "HEAD", "OPTIONS"}
+
+
+def get_csrf_from_request(request: Request) -> tuple[str | None, str | None]:
+    """Extract CSRF token from cookie and form body is handled in routes."""
+    cookie_val = request.cookies.get(CSRF_COOKIE)
+    return cookie_val
```

Ruff (0.15.9)

[warning] Unnecessary assignment to `cookie_val` before `return` statement

Remove unnecessary assignment

(RET504)